service/history/queues/executable.go

```diff
@@ -517,21 +529,22 @@ func (e *executableImpl) HandleErr(err error) (retErr error) {
 	if ok, rewrittenErr := e.isExpectedRetryableError(err); ok {
 		return rewrittenErr
 	}
+
 	// Unexpected errors handled below
 	e.unexpectedErrorAttempts++
-	metrics.TaskFailures.With(e.taggedMetricsHandler).Record(1)
+	metrics.TaskFailures.With(metricsHandler).Record(1)
 	e.logger.Warn("Fail to process task", tag.Error(err), tag.ErrorType(err), tag.UnexpectedErrorAttempts(int32(e.unexpectedErrorAttempts)), tag.LifeCycleProcessingFailed)
 
 	if e.isUnexpectedNonRetryableError(err) {
```

Comment:
we can remove TaskInternalErrorCounter inside this func as well, now that we can always get this info from TaskFailures metric.

Reply:
TaskFailures doesn't have any tags with the metrics.

Reply:
I thought you want to add error type tag to TaskFailures metric?

Reply:
But if I only add the cause tag to TaskThrottledCounter, I don't need to add the error type to TaskFailures.

Reply:
Sure. I am asking because from the PR title and desc, it seems like you want both, hence the question.